mpf/config_players/segment_display_player.py

```diff
@@ -106,12 +106,23 @@ def _remove(self, instance_dict, key, display):
             del instance_dict[display][key]
 
     def clear_context(self, context):
-        """Remove all texts."""
-        instance_dict = self._get_instance_dict(context)
-        for display, keys in instance_dict.items():
-            for key in dict(keys).keys():
-                self._remove(instance_dict=instance_dict,
-                             key=key, display=display)
+
+        ##############
+        # Remove all texts. Ignore what keys are available, that will be checked later in the segment display code.
+        # Especially important for update_method replace since there are no keys.
+        ##############
+
+        instance_dict = self._get_instance_dict(context) # key of the dict is the display, the value is another dict
+
+        for display, keys_dict in instance_dict.items(): # keys_dict key is the show key, the value is a boolean (with yet unknown usage)
+            if(keys_dict): #depending on the situation the keys_dict might be empty, still need to clear the display
+                for key in dict(keys_dict).keys():
```

Comment:
In what situation would keys_dict *not* be a dict? It feels weird to convert it to dict here instead of `for key in keys_dict.keys()`